qa/rpc-tests/p2p_txexpiringsoon.py

```diff
@@ -6,111 +6,81 @@
 import sys; assert sys.version_info < (3,), ur"This script does not run under Python 3. Please use Python 2.7.x."
 
 from test_framework.authproxy import JSONRPCException
-from test_framework.mininode import NodeConn, NodeConnCB, NetworkThread, \
-    CTransaction, CInv, msg_mempool, msg_getdata, msg_tx, mininode_lock, \
-    msg_ping, msg_pong, OVERWINTER_PROTO_VERSION
+from test_framework.mininode import NodeConn, NetworkThread, CInv, \
+    msg_mempool, msg_getdata, msg_tx, mininode_lock, OVERWINTER_PROTO_VERSION
 from test_framework.test_framework import BitcoinTestFramework
-from test_framework.util import initialize_chain_clean, start_nodes, \
-    p2p_port, assert_equal, sync_blocks, sync_mempools, connect_nodes_bi
-
-import time, cStringIO
-from binascii import hexlify, unhexlify
-
-class TestNode(NodeConnCB):
-    def __init__(self):
-        NodeConnCB.__init__(self)
-        self.create_callback_map()
-        self.connection = None
-        self.ping_counter = 1
-        self.last_pong = msg_pong()
-
-    def add_connection(self, conn):
-        self.connection = conn
-
-    # Spin until verack message is received from the node.
-    # We use this to signal that our test can begin. This
-    # is called from the testing thread, so it needs to acquire
-    # the global lock.
-    def wait_for_verack(self):
-        while True:
-            with mininode_lock:
-                if self.verack_received:
-                    return
-            time.sleep(0.05)
-
-    # Wrapper for the NodeConn's send_message function
-    def send_message(self, message):
-        self.connection.send_message(message)
-
-    def on_close(self, conn):
-        pass
-
-    def on_reject(self, conn, message):
-        conn.rejectMessage = message
-
-    # Track the last getdata message we receive (used in the test)
-    def on_getdata(self, conn, message):
-        self.last_getdata = message
-
-    def on_tx(self, conn, message):
-        self.last_tx = message
-
-    def on_inv(self, conn, message):
-        self.last_inv = message
-
-    def on_notfound(self, conn, message):
-        self.last_notfound = message
-
-    def on_pong(self, conn, message):
-        self.last_pong = message
-
-    # Sync up with the node after delivery of a message
-    def sync_with_ping(self, timeout=30):
-        self.connection.send_message(msg_ping(nonce=self.ping_counter))
-        received_pong = False
-        sleep_time = 0.05
-        while not received_pong and timeout > 0:
-            time.sleep(sleep_time)
-            timeout -= sleep_time
-            with mininode_lock:
-                if self.last_pong.nonce == self.ping_counter:
-                    received_pong = True
-        self.ping_counter += 1
-        return received_pong
+from test_framework.util import assert_equal, connect_nodes_bi, fail, \
+    initialize_chain_clean, p2p_port, start_nodes, sync_blocks, sync_mempools
+from tx_expity_helper import TestNode, create_transaction
```

Comment:
Typo in helper module name: `tx_expiry_helper`

Reply:
Whoops... Thanks for catching this. This is now fixed.